modules/ROOT/pages/common-commands-troubleshoot.adoc

```diff
@@ -0,0 +1,96 @@
+= Common Commands Used for Troubleshooting
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: service mesh, microservices, common commands, troubleshooting
+
+When troubleshooting Anypoint Service Mesh, you might need to perform various common tasks, such as retrieving adapter logs, displaying the adapter status, modifying log4j configuration, and so on:
+
+* <<retrieve-adapter-logs,Retrieve adapter logs>>
+* <<access-the-adapter,Access the adapter>>
+* <<retrieve-policies-and-contracts-info,Retrieve policies and contract information from the adapter>>
+* <<modify-log4j-config,Modify log4j configuration>>
+* <<replace-service-mesh-license,Replace Anypoint Service Mesh license>>
+
+[[retrieve-adapter-logs]]
+== Retrieve Adapter Logs
+
+To retrieve the adapter logs when you are troubleshooting an issue:
+
+. Display the adapter pod:
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text of the screenshot here
+. Retrieve the adapter logs:
+`$ kubectl -n service-mesh logs $(kubectl -n service-mesh get pod -l namespace=<replace_with_namespace> -oname) -c mule`.
+
+[[access-the-adapter]]
+== Access the Adapter
+
+When troubleshooting Anypoint Service Mesh, you might need to access and display the information about the adapter.
+
+//[PA: Kim is this format better or the ones that I follow in the next few topics?]
+To display information about the adapter pods:
+
+. Dispalys all the pods created under the namespace service-mesh:
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text for the screenshot here
+. Access the adapter:
++
+`$ kubectl -n service-mesh exec -ti  <adapter-pod-name>`.
+
+[[retrieve-policies-and-contracts-info]]
+== Retrieve Policies and Contracts Information from the Adapter
+
+When troubleshooting issues with Anypoint Service Mesh, you might need to retrieve policies and contracts details:
+
+. Display the adapter pod: 
++
+`$ kubectl get pods -n service-mesh`.
++
+An output is displayed that looks like this:
++
+//add text for the screenshot here
+. Download the policy and contract files:
++
+`$ kubectl cp service-mesh/$(kubectl -n service-mesh get pod -ocustom-columns='NAME:.metadata.name' --no-headers -l namespace=<replace_with_namespace>):/usr/src/app/policies -c mule ~/Desktop/`.
+
+[[modify-log4j-config]]
+== Modify Log4j configuration
+
+When troubleshooting Anypoint Service Mesh, you might need to modify the Log4j configuration:
+
+. Execute the following command to display the adapter pod:
```

Comment:
These tasks are so short: mostly two steps, and 4 of 5 begin with the very same command.
I'm wondering if there isn't a way to avoid repeating that same Step 1 and MISSING output display.
Maybe we can brainstorm this later.

Reply:
Yes, sure! Done! Thanks!